spark-cassandra-connector/src/main/scala/org/apache/spark/sql/cassandra/DefaultSource.scala

```diff
@@ -0,0 +1,93 @@
+package org.apache.spark.sql.cassandra
+
+import org.apache.spark.sql.{DataFrame, SaveMode, SQLContext}
+import org.apache.spark.sql.SaveMode._
+import org.apache.spark.sql.sources.{CreatableRelationProvider, SchemaRelationProvider, BaseRelation, RelationProvider}
+import org.apache.spark.sql.types.{DataType, StructType}
+
+
+class DefaultSource extends RelationProvider with SchemaRelationProvider with CreatableRelationProvider {
+
+  /** Creates a new relation for a cassandra table given table, keyspace, and cluster as a parameter.*/
+  override def createRelation(sqlContext: SQLContext,
+                              parameters: Map[String, String]): BaseRelation = {
+    val schemaJsonString = getUserDefinedSchema(parameters)
+    val schema = if (schemaJsonString == None) None
+    else Option(DataType.fromJson(schemaJsonString.get).asInstanceOf[StructType])
```

Comment:
and I don't like the fact, that there is `asInstanceOf` here - are you sure it will always be a `StructType`? If so, why Spark API returns a more general `DataType`?